### PR TITLE
[#2380] Clicking an image with a link only opens on exported projects

### PR DESCRIPTION
**Pull request by dseif**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/741ffb5c666c2d1/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/93618275ab58ba6/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/741ffb5c666c2d1/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/93618275ab58ba6/output.txt

Total script time: 3.00 mins

File: templates/assets/editors/image/image-editor.js [hunk 1]
```diff
@@ -420,10 +420,17 @@
...
+          var link;
+
           _trackEvent = e.target;
           calcImageTime();
           _this.updatePropertiesFromManifest( _trackEvent );
 
+          link = _trackEvent.popcornTrackEvent._container.querySelector( "a" );
+          link.addEventListener( "click", function( e ) {
+            e.preventDefault();
+          }, false);
```
Comment (mjschranz) [inline templates/assets/editors/image/image-editor.js]: Space.

Comment (dseif) [inline templates/assets/editors/image/image-editor.js]: huh

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/419e249866337a1/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/78ad5f3bd40c6f5/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/419e249866337a1/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/78ad5f3bd40c6f5/output.txt

Total script time: 3.00 mins

File: templates/assets/editors/image/image-editor.js [hunk 2]
```diff
@@ -381,7 +381,14 @@
...
         var popcornOptions = trackEvent.popcornOptions,
-            manifestOpts = trackEvent.popcornTrackEvent._natives.manifest.options;
+            manifestOpts = trackEvent.popcornTrackEvent._natives.manifest.options,
+            link = trackEvent.popcornTrackEvent._container.querySelector( "a" );
+
+        function linkClick( e ) {
+          e.preventDefault();
+        }
+
+        link.addEventListener( "click", linkClick, false );
```
Comment (mjschranz) [inline templates/assets/editors/image/image-editor.js:391]: What is link is undefined here? I'm surprised I'm not getting console errors on opening a editor for a new image.

File: templates/assets/editors/image/image-editor.js [hunk 3]
```diff
@@ -421,9 +428,12 @@
 
         _trackEvent.listen( "trackeventupdated", function( e ) {
           _trackEvent = e.target;
+          link = _trackEvent.popcornTrackEvent._container.querySelector( "a" );
           calcImageTime();
           _this.updatePropertiesFromManifest( _trackEvent );
 
+          link.addEventListener( "click", linkClicked, false );
```
Comment (mjschranz) [inline templates/assets/editors/image/image-editor.js:435]: Again, what if link is undefined? I think we need safer checks here.

**Comment by mjschranz** [comment]
#1110 is now covering this.

File: templates/assets/editors/image/image-editor.js [hunk 4]
```diff
@@ -381,7 +381,14 @@
     Editor.TrackEventEditor.extend( _this, butter, rootElement, {
       open: function( parentElement, trackEvent ) {
         var popcornOptions = trackEvent.popcornOptions,
-            manifestOpts = trackEvent.popcornTrackEvent._natives.manifest.options;
+            manifestOpts = trackEvent.popcornTrackEvent._natives.manifest.options,
+            link = trackEvent.popcornTrackEvent._container.querySelector( "a" );
+
+        function linkClick( e ) {
+          e.preventDefault();
```
Comment (ScottDowne) [inline templates/assets/editors/image/image-editor.js:388]: You can do a return false here to cancel the functionality too. This way you can still select the text. preventDefault would prevent other things not ust the click. May cause unexpected behaviour.

Comment (ScottDowne) [inline templates/assets/editors/image/image-editor.js:388]: Oh, I htink in order to use return false, it needs to be applied with 

elem.onclick = function() { return false; };

**Pull request closed without merging**